packages/core/src/collections/map/bucket_map.candy

```diff
@@ -7,66 +7,95 @@ use ...iterable
 use ...list
 use ...map
 
+// TODO(marcelgarus): Make the `BucketMap` implementation way more efficient.
+
+fun hashByKey(key: Hash): Int { key.hash<Int>(Hasher.default_()) }
+
 class MapEntry<Key, Value> {
   let key: Key
   let hash: Int
-  let value: Value
+  mut let value: Value
 }
 
 class Bucket<Key: Equals & Hash, Value> {
-  let entries: List<MapEntry<Key, Value>>
+  /// An inefficient "mini-map" (it's typically pretty small).
```

Comment:
Maybe we should expose this class as a `LinkedHashMap`, or at least keep that in mind when we want to offer different map types in the future.